src/rustup-dist/src/errors.rs

```diff
@@ -127,11 +127,25 @@ fn component_unavailable_msg(cs: &[Component], manifest: &Manifest) -> String {
         if same_target {
             let mut cs_strs = cs.iter().map(|c| format!("'{}'", c.short_name(manifest)));
             let cs_str = cs_strs.join(", ");
-            let _ = write!(buf, "some components unavailable for download: {}", cs_str);
+            let _ = write!(
+                    buf,
+                    "some components unavailable for download: {}\n\
+                    if you want these components, please install and use latest successful build version, \
+                    which you can find out at https://rust-lang-nursery.github.io/rust-toolstate, like this.\n\
```

Comment:
I'd drop the 'out' replace the ',' with '.' and change 'like this.' to '\nTo do this, run something like:\n'